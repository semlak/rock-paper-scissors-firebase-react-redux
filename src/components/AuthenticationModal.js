import React from 'react';
import {
  Modal,
  // ModalHeader,
  ModalBody,
  // ModalFooter,
  Button,
  TabContent,
  TabPane,
  Nav,
  NavItem,
  NavLink,
  Row,
  Col,
  ButtonGroup,

} from 'reactstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { signIn, signInWithGithub, loginUserWithEmailPassword, registerUserAction } from '../actions/user';
import { toggleAuthenticationModal, modalMessage } from '../actions/modalActions';
import { LoginForm, RegistrationForm } from '../components';

class AuthenticationModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      activeTab: '1',
    };
  }

  toggleTab(tab) {
    if (this.state.activeTab !== tab) {
      this.setState({
        activeTab: tab
      });
    }
  }

  render() {
    // console.log('rendingering AuthenticationModal, props:', this.props);
    const { signIn: googleSignIn, signInWithGithub: githubSignIn, authenticationModalOpen: isOpen, toggleAuthenticationModal: toggle } = this.props;

    return (
      <Modal isOpen={isOpen} toggle={toggle} size="lg" className={this.props.className}>
        <ModalBody>
          <Row className="row">
            {/* <div className="col-lg-8 col-md-12 add-right-border"> */}
            <Col lg={8} md={12} className="add-right-border">
              {/* <!-- Nav tabs --> */}
              {/* <!-- modified from https://bootsnipp.com/snippets/VElzQ --> */}
              <Nav tabs>
                <NavItem>
                  <NavLink
                    className={this.state.activeTab === '1' ? 'active' : ''}
                    onClick={() => { this.toggleTab('1'); }}
                  >
                    Login
                  </NavLink>
                </NavItem>
                <NavItem>
                  <NavLink
                    className={this.state.activeTab === '2' ? 'active' : ''}
                    onClick={() => { this.toggleTab('2'); }}
                  >
                    Registration
                  </NavLink>
                </NavItem>
              </Nav>
              <TabContent activeTab={this.state.activeTab} className="mt-2" id="nav-tab-content">
                <TabPane tabId="1">
                  <LoginForm closeButtonAction={toggle} {...this.props} />
                </TabPane>
                <TabPane tabId="2">
                  <RegistrationForm closeButtonAction={toggle} {...this.props} />
                </TabPane>
              </TabContent>
              {/* <!-- Sign in with Google of Github--> */}
              <div id="OR" className="">OR</div>
            </Col>
            {/* <div className="hidden-lg-down col-sm-12 col-md-12 col-lg-4"> */}
            <Col sm={12} md={12} lg={4} className="hidden-lg-down">
              <Row className="text-center sign-with">
                <Col md={12}>
                  <h5>Sign in with</h5>
                </Col>
                <Col md={12}>
                  {/* <div className="btn-group btn-group-justified"> */}
                  <ButtonGroup>
                    <Button id="github-signin" onClick={githubSignIn}>Github</Button>
                    <Button color="danger" id="google-signin" onClick={googleSignIn}>Google</Button>
                  </ButtonGroup>
                </Col>
              </Row>
            </Col>
          </Row>
        </ModalBody>
      </Modal>
    );
  }
}


function mapStateToProps({ modals }) {
  const { authenticationModalOpen } = modals;
  return { authenticationModalOpen, modals };
}


AuthenticationModal.defaultProps = {
};

AuthenticationModal.propTypes = {
  // auth: PropTypes.shape({
  //   displayName: PropTypes.string,
  //   uid: PropTypes.string,
  //   photoURL: PropTypes.string,
  // }),
  signIn: PropTypes.func.isRequired,
  // signOut: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, { signIn, signInWithGithub, toggleAuthenticationModal, loginUserWithEmailPassword, registerUserAction, modalMessage })(AuthenticationModal);
