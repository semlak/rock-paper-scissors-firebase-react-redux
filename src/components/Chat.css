/* chat */

.col-sm-1.order-sm-1.offset-sm-0 {
  /* padding-right: 0; */
}

.col-sm-1.order-sm-2.offset-sm-0 {
  padding-left: 0;
  padding-right: 25;
}

.sent_msg.col-sm-6.order-sm-1.offset-sm-5 {
  /* margin-right: 2px; */
}

.received_withd_msg .card-body {
  /* background: #ebebeb none repeat scroll 0 0; */
  /* border: none; */
  /* border: solid black 0px; */
  /* border-radius: 3px; */
  color: #646464;
  /* font-size: 14px; */
  /* margin: 0; */
  padding: 5px 10px 5px 12px;
  /* width: 100%; */
}


.card-footer.time_date {
  color: #747474;
  background: #fff;
  display: block;
  padding-left: 0;
  padding-top: .2em;
  border: none;
  /* border-style: none; */
  font-size: 12px;
  /* margin: 8px 0 0; */
}

.mesgs {
  /* float: left; */
  /* padding: 30px 15px 0 25px; */
  padding: 0px 15px 0 25px;
  margin-top: -10px;
  /* width: 60%; */
  width: 100%;
}

.message.card-body {
  border-radius: 10px;
}

.sent_msg .card-body {
  /* background: #05728f none repeat scroll 0 0; */
  /* border-radius: 3px; */
  /* font-size: 14px; */
  /* margin: 0;  */
  color:#fff;
  padding: 5px 10px 5px 12px;
  width:100%;
}


.msg_send_btn {
  /* background: #05728f none repeat scroll 0 0; */
  /* background: #05728f; */
  border-top-right-radius: 50%;
  border-bottom-right-radius: 50%;
  /* border: medium none; */
  /* border-radius: 50%; */
  color: #fff;
  cursor: pointer;
}

.messaging {
  padding: 0 0 50px 0;
}

.msg_history {
  /* height: 516px; */
  height: 516px;
  overflow-y: auto;
  overflow-x: hidden;
  margin-right: -0.5rem;
  /* overflow: hidden; */
  /* overflow: auto; */
  margin-top: 1rem;
}

.msg_history > .row {
  margin-right: .3rem;
}
