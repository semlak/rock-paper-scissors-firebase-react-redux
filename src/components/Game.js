import React, { Component } from 'react';
import {
  Button,
  Card,
  CardBody,
  // CardTitle,
  CardText,
  CardFooter,
  CardHeader,
} from 'reactstrap';
import { connect } from 'react-redux';

// import firebase from '../firebase';
import LoadingSpinner from './LoadingSpinner';
import {
  endGame,
  makePlay,
  getRoundOutcome,
} from '../actions/game';
import gameStatuses from '../gameStatuses';


class Game extends Component {
  state = this.getInitialState();

  getInitialState() {
    const { user } = this.props;
    const {
      // gameInProgress,
      // maxNumberOfGames,
      player1: player1UID,
      player2: player2UID,
      player1Wins,
      player2Wins,
      player1Name,
      player2Name,
      ties,
      playInProgress,
    } = this.props.game.gameData;
    // const { uid: myUID, displayName: myName } = user;
    const { uid: myUID, } = user;
    // console.log(`player1UID ${player1UID}, player2UID ${player2UID}, myUID ${myUID}, player1UID === myUID: ${player1UID === myUID}`);
    const player1 = { uid: player1UID, name: player1Name, wins: player1Wins };
    const player2 = { uid: player2UID, name: player2Name, wins: player2Wins };
    // const { uid: opponentUID, name: opponentName, wins: opponentWins } = player1UID === myUID ? player2 : player2UID === myUID ? player1 : null;
    const { name: opponentName, wins: opponentWins } = player1UID === myUID ? player2 : player2UID === myUID ? player1 : {};
    const { wins: myWins } = player1UID === myUID ? player1 : player2;
    // const player1or2 = player1UID === myUID ? "player1" : "player2";

    return ({
      // myUID,
      // opponentUID,
      opponentName,
      myWins,
      opponentWins,
      ties,
      // maxNumberOfGames,
      // gameInProgress,
      // player1or2,
      playInProgress,
    });
  }

  componentDidUpdate(prevProps) {
    const { gameStatus } = this.props.game;
    if (gameStatus === gameStatuses.DETERMINING_ROUND_WINNER && gameStatus !== prevProps.game.gameStatus && prevProps.game.gameStatus.length > 0) {
      this.props.getRoundOutcome(this.props.gameUID);
    }
  }

  play = (e) => {
    const { value: playerAction } = e.target;
    // const player = playerid === game.player1 ? "player1" : "player2";
    const { player1: player1UID, player2: player2UID } = this.props.game.gameData;
    const { uid: myUID } = this.props.user;
    const player1or2 = player1UID === myUID ? "player1" : player2UID === myUID ? "player2" : "";
    // const { player1or2 } = this.state;
    // console.log('e.target', e.target, 'playerAction ', playerAction);
    // console.log('player1UID', player1UID, 'myUID', myUID, 'player1or2', player1or2, 'playerAction', playerAction);
    if (playerAction.match(/(rock|paper|scissors)/) != null && player1or2 && player1or2.length > 1) {
      const { gameKey } = this.props.game;
      // this.props.makePlay(gameKey, player1or2, playerAction);
      this.props.makePlay(gameKey, player1or2, myUID, playerAction);
    }
    else {
      console.log('invalid playerAction or playerinfo', playerAction, player1or2);
    }
  }

  endCurrentGame = () => {
    console.log('ending current game');
    if (this.props.endGame && this.props.gameUID) {
      console.log('creating endGame action');
      this.props.endGame(this.props.gameUID);
    }
  }

  render = () => {
    const {
      opponentName,
      // myUID,
      // opponentUID,
      myWins,
      opponentWins,
      ties,
      // messages,
      // maxNumberOfGames,
      // gameInProgress,
      // playInProgress,
    } = this.state; 

    const { gameStatus } = this.props.game;
    // const playEnabled = gameStatus !== gameStatuses.PLAY_MADE_WAITING_FOR_OPPONENT && gameStatus !== gameStatuses.DETERMINING_ROUND_WINNER;
    const playEnabled = true;


    // const loadingMessage = 'You played ...';
    const loadingMessage = gameStatus;

    // const loadingMessageClass = '
    const loadingMessageStyle = { textAlign: 'center', marginTop: '1em', size: '22px' };
    // const { loading, file } = this.state;
    const playInProgress = !playEnabled;
    // playInProgress = false;
    const loading = playInProgress;
    const spinnerProps = {
      // loadingMessage,
      loadingMessageStyle,
      loading,
      className: 'loading-viewer',
      // delay: 500,
      // fadeIn: true,
      id: "loading-spinner",
    };
    // console.log('spinnerProps', spinnerProps);


    const activeCardBody = (
      <CardBody className="">
        <p> Your Opponent: <span id="opponent-display-name">{opponentName || "Loading..."}</span></p>
        <p>
          Your score: <span id="your-score">{myWins}</span>
        </p>
        <p>
          Opponent Score: <span id="opponent-score">{opponentWins}</span>
        </p>
        <p>
          Ties: <span id="ties">{ties}</span>
        </p>
        { playInProgress ? <LoadingSpinner {...spinnerProps} /> :
            <>
              <Button color="primary" className="mr-2 game-action" onClick={this.play} value="rock" disabled={!playEnabled}> Rock </Button>
              <Button color="primary" className="mr-2 game-action" onClick={this.play} value="paper" disabled={!playEnabled}> Paper </Button>
              <Button color="primary" className="mr-2 game-action" onClick={this.play} value="scissors" disabled={!playEnabled}> Scissors </Button>
              <br />
              <br />
            </>
        }
        <Button color="danger" onClick={this.endCurrentGame} id="end-game">End Game</Button>
      </CardBody>
    );
    // return <LoadingSpinner loading/>


    return (
      <Card className="border-success game" style={{ display: "block" }} >
        <CardHeader className="text-white bg-success bg">
          Your Active Game
        </CardHeader>
        {/* {playInProgress ? <LoadingSpinner {...spinnerProps} /> : activeCardBody} */}
        {activeCardBody}
        <CardFooter><CardText>{loadingMessage}</CardText></CardFooter>
      </Card>
    );
  }
}


function mapStateToProps({ auth, gathering, game }) {
  const onlinePlayers = gathering != null ? gathering.onlinePlayers || [] : [];
  return {
    user: auth,
    gathering,
    players: onlinePlayers,
    game,
    gameUID: game.gameKey,
  };
}


export default connect(mapStateToProps, { endGame, makePlay, getRoundOutcome })(Game);
