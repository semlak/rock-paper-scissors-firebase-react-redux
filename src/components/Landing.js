import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Row, Col, Card, Button } from 'reactstrap';

import { toggleAuthenticationModal } from '../actions/modalActions';
import "./Landing.css";

import landingImage from '../img/landing-rps.jpg';

export class Landing extends Component {
  static contextTypes = {
    router: PropTypes.object
  };

  componentWillUpdate(nextProps) {
    if (nextProps.auth) {
      this.context.router.history.push("/app");
    }
  }

  toggleModal = (e) => {
    e.preventDefault();
    this.props.toggleAuthenticationModal();
  }

  render() {
    return (
      <div>
        <Row className="social-signin-container">
          <Col>
            {/* <img alt="Sign in" id="sign-in" src="/img/user.png" /> */}
            <img alt="Rock Paper Scissors" id="sign-in" width="100%" src={landingImage} />
            <br />
            <br />
            <br />
            <br />
            {/* <h4 id="sign-in-header">Sign In to start</h4> */}
            <Button href="/#" className="social-signin" onClick={this.toggleModal}>
              Sign in to start
            </Button>
            {/* <a href="/#" className="social-signin" onClick={this.toggleModal}> */}
            {/*   Sign in to start                                                 */}
            {/* </a>                                                               */}
          </Col>
        </Row>
        <Row className="mt-4 mb-4 text-left">
          <Col >
            <Card body outline color="danger">
              <p>For trying out the game with multiple players, you can use separate logins in separate browsing sessions.</p>
              <p>You can create your own logins or use the following sample logins (each with password &apos;<samp>password</samp>&apos;):</p>
              <ul>
                <li><samp>harry@hogwarts.co.uk</samp></li>
                <li><samp>ron@hogwarts.co.uk</samp></li>
                <li><samp>hermione@hogwarts.co.uk</samp></li>
              </ul>
            </Card>
          </Col>
        </Row>
      </div>
    );
  }
}

function mapStateToProps({ auth }) {
  return { auth };
}

export default connect(mapStateToProps, { toggleAuthenticationModal })(Landing);
