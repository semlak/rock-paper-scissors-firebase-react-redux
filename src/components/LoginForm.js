import React from 'react';
import {
  // ModalHeader,
  // ModalFooter,
  Button,
  // Container,
  Row,
  Col,
  Input,
  Form,
  FormGroup,
  FormFeedback,
  Label,
} from 'reactstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { loginUserWithEmailPassword, } from '../actions/user';


class AuthenticationModal extends React.Component {
  constructor(props) {
    super(props);
    // this.handleInputChange = this.handleInputChange.bind(this);
    this.state = {
      email: '',
      password: '',
      // formValid: false,
      // formInvalid: false,
    };
  }

  handleInputChange = event => this.setState({ [event.target.name]: event.target.value });

  handleLoginClick = event => {
    event.preventDefault();
    // const { registerUser } = this.props;
    console.log('in handleRegisterClick');
    // const { props } = this;
    const { email, password, } = this.state;
    if (password.length >= 6) {
      this.props.loginUserWithEmailPassword({ email, password });
    }
    else {
      console.error('password needs to be at least 6 characters');
    }
  }

  render() {
    // const { user, login, logout } = this.props;
    // const { props } = this;
    // const { auth, signIn: userSignIn, signOut: userSignOut } = this.props;
    // const { auth, signIn: googleSignIn, signOut: signOutAction, gathering } = this.props;
    const { email, password } = this.state;
    // console.log('gathering in Navbar component:', gathering);
    // const userSignOut = () => signOutAction(gathering);
    // const userSignOut = () => signOutAction({ auth, gathering });
    // const userSignOut = () => signOutAction(gathering);
    // console.log('auth', auth);
    const { handleInputChange } = this;
    console.log('this.props', this.props);
    const { loginError: modalError, message } = this.props.modals;
    const valid = !modalError && typeof modalError !== 'undefined' ? true : undefined;
    const invalid = !!modalError && typeof modalError !== 'undefined' ? true : undefined;
    // console.log('invalid:', invalid);

    return (
      <Form >
        <FormGroup row >
          <Label for="email" sm={3} >Email</Label>
          <Col sm={9}>
            <Input valid={valid} invalid={invalid} type="email" name="email" value={email} placeholder="Email" onChange={handleInputChange} />
          </Col>
        </FormGroup>
        <FormGroup row >
          <Label for="password" sm={3} >Password</Label>
          <Col sm={9}>
            <Input valid={valid} invalid={invalid} type="password" name="password" value={password} placeholder="password" onChange={handleInputChange} />
            <FormFeedback valid={valid} invalid={typeof invalid !== 'undefined' ? 'false' : undefined} >{message}</FormFeedback>
          </Col>
          {/* <FormFeedback valid={modalError}>Logging in. Please wait...</FormFeedback> */}
        </FormGroup>
        <Row>
          <Col sm={{ size: 9, offset: 3 }} >
            <Button color="primary" onClick={this.handleLoginClick}>Submit</Button>
            <Button color="danger" onClick={this.props.closeButtonAction} className="ml-2" data-dismiss="modal">Cancel</Button>
          </Col>
        </Row>
      </Form>
    );
  }
}

// function mapStateToProps({ auth }) {
//   return { auth };
// }


// AppNavbar.defaultProps = {
//   auth: {},
// };


function mapStateToProps({ auth, gathering, modals }) {
  return { auth, gathering, modals };
}


AuthenticationModal.defaultProps = {
  // auth: {},
};

AuthenticationModal.propTypes = {
  // auth: PropTypes.shape({
  //   displayName: PropTypes.string,
  //   uid: PropTypes.string,
  //   photoURL: PropTypes.string,
  // }),
  closeButtonAction: PropTypes.func.isRequired,
  loginUserWithEmailPassword: PropTypes.func.isRequired,
  // signOut: PropTypes.func.isRequired,
};

// export default connect(mapStateToProps, { signIn, signOut })(AuthenticationModal);
export default connect(mapStateToProps, { loginUserWithEmailPassword, })(AuthenticationModal);
