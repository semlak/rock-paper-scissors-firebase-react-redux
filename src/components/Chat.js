import React, { Component } from 'react';
import {
  Button,
  // Container,
  Row,
  Col,
  Card,
  CardBody,
  // CardTitle,
  // CardText,
  CardFooter,
  CardHeader,
} from 'reactstrap';
import { connect } from 'react-redux';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPaperPlane } from '@fortawesome/free-solid-svg-icons';
import Avatar from 'react-avatar';
import moment from 'moment';

// import { Widget, addResponseMessage, addLinkSnippet, addUserMessage } from 'react-chat-widget';
// import 'react-chat-widget/lib/styles.css';
import './Chat.css';

// import Avatar from 'react-avatar';

// import defaultPlayerImage from '../img/user.png';

// import { joinGathering } from '../actions/gatheringActions';
import {
  changePlayerStatus,
} from '../actions/gatheringActions';


const PlayerAvatar = ({ player }) => (
  player.photoURL && player.photoURL.length > 0 ?
    <img src={player.photoURL} height="37" alt={player.displayName} /> :
    <Avatar name={player.displayName} color={Avatar.getRandomColor('sitebase', ['green'])} size="37" textSizeRatio={2} round={true} />
);

const getTimeFromStamp = (timestamp) => {
  // const d = new Date(timestamp);
  const d = moment(timestamp);
  // timeStampCon = d.getDate() + '/' + (d.getMonth()) + '/' + d.getFullYear() + " " + d.getHours() + ':' + d.getMinutes();
  // const timeStampCon = d.getHours() + ':' + d.getMinutes();
  const timeStampCon = d.format("h:mm A");
  return timeStampCon;
};


const getDateFromStamp = (timestamp) => {
  // const d = new Date(timestamp);
  const d = moment(timestamp);
  // timeStampCon = d.getDate() + '/' + (d.getMonth()) + '/' + d.getFullYear() + " " + d.getHours() + ':' + d.getMinutes();
  console.log('d', d);
  // const dateStampCon = (d.getMonth()) + '/' + d.getDate();
  const dateStampCon = d.format("MMMM DD");
  // d.getFullYear() + " " + d.getHours() + ':' + d.getMinutes();
  // timeStampCon = d.getHours() + ':' + d.getMinutes();
  return dateStampCon;
};

const Message = ({ direction, text, timestamp, imgSrc, name }) => {
  
  const time = getTimeFromStamp(timestamp);
  const date = getDateFromStamp(timestamp);

  return (
    <Row className={direction === 'incoming' ? 'incoming_msg' : 'outgoing_msg'}>
      <Col sm={{ size: 1, order: direction === 'incoming' ? 1 : 2, offset: 0 }} >
        <PlayerAvatar player={{ photoURL: imgSrc, displayName: name }} />
      </Col>
     

      {/* <Col className="incoming_msg_img1">                     */}
      {/*   <img src={imgSrc} className="img-fluid" alt={name} /> */}
      {/* </Col>                                                  */}
      <Col sm={{ size: 6, order: direction === 'incoming' ? 2 : 1, offset: direction === 'incoming' ? 0 : 5 }} className={direction === 'incoming' ? 'received_msg' : 'sent_msg'}>
        <Card className={direction === 'incoming' ? 'received_withd_msg1' : 'sent_withd_msg1'}>
          <CardBody>{text}</CardBody>
          <CardFooter className="time_date">{time} | {date}</CardFooter>
        </Card>
      </Col>

      {/* { imgSrc ? <div className={`${direction}_msg_img`}><img src={imgSrc} alt={name} /> </div> : "" } */}
      {/* <div className={direction === 'incoming' ? 'received_msg' : 'sent_msg'}> */}
      {/*   <div className={direction === 'incoming' ? 'received_withd_msg' : ''}> */}
      {/*     <p>{text}</p>                                                        */}
      {/*     <span className="time_date">{time} | {date}</span>                   */}
      {/*   </div>                                                                 */}
      {/* </div>                                                                   */}
    </Row>
  );
};

// const Message = ({ direction, text, timestamp, imgSrc, name }) => {
//   const time = getTimeFromStamp(timestamp);
//   const date = getDateFromStamp(timestamp);
//   return (
//     <div className={direction === 'incoming' ? 'incoming_msg' : 'outgoing_msg'}>
//       <PlayerAvatar player={{ photoURL: imgSrc, displayName: name }} />
//       [> { imgSrc ? <div className={`${direction}_msg_img`}><img src={imgSrc} alt={name} /> </div> : "" } <]
//       <div className={direction === 'incoming' ? 'received_msg' : 'sent_msg'}>
//         <div className={direction === 'incoming' ? 'received_withd_msg' : ''}>
//           <p>{text}</p>
//           <span className="time_date">{time} | {date}</span>
//         </div>
//       </div>
//     </div>
//   );
// };

// const UserList = (props) => {
class Chat extends Component {
  componentDidMount() {
  }

  componentDidUpdate() {
  }


  handleNewUserMessage = (newMessage) => {
    console.log(`New message incoming! ${newMessage}`);
    // Now send the message throught the backend API
    // addResponseMessage(newMessage);
  }

  render() {
    const messages = [{
      direction: 'incoming',
      timestamp: moment('06/09/2018 11:01'),
      text: "Test which is a new approach to have all solutions",
      imgSrc: "https://ptetutorials.com/images/user-profile.png",
      name: "sunil",
    }, {
      direction: 'outgoing',
      timestamp: moment('06/09/2018 11:00'),
      text: "Test which is a new approach to have all solutions",
      // imgSrc: "https://ptetutorials.com/images/user-profile.png",
      name: "xena",
    }];

    const blah = (
      <Row>
        <div className="mesgs">
          <div className="msg_history">
            {messages.map(message => <Message key={`${message.text}_${message.timestamp}`} {...message} />)}
            {/* <Message                                                    */}
            {/*   direction="incoming"                                      */}
            {/*   // timestamp={1552603967218}                              */}
            {/*   timestamp={moment('06/09/2018 11:01')}                    */}
            {/*   text="Test which is a new approach to have all solutions" */}
            {/*   imgSrc="https://ptetutorials.com/images/user-profile.png" */}
            {/*   name="sunil"                                              */}
            {/* />                                                          */}
            {/* <div className="incoming_msg">                                                                                  */}
            {/*   <div className="incoming_msg_img"> <img src="https://ptetutorials.com/images/user-profile.png" alt="sunil" /> */}
            {/*   </div>                                                                                                        */}
            {/*   <div className="received_msg">                                                                                */}
            {/*     <div className="received_withd_msg">                                                                        */}
            {/*       <p>Test which is a new approach to have all solutions</p>                                                 */}
            {/*       <span className="time_date"> 11:01 AM    |    June 9</span>                                               */}
            {/*     </div>                                                                                                      */}
            {/*   </div>                                                                                                        */}
            {/* </div>                                                                                                          */}
            <div className="outgoing_msg">
              <div className="sent_msg">
                <p>Test which is a new approach to have all solutions</p>
                <span className="time_date"> 11:01 AM    |    June 9</span>
              </div>
            </div>
            <div className="incoming_msg">
              <div className="incoming_msg_img"> <img src="https://ptetutorials.com/images/user-profile.png" alt="sunil" />
              </div>
              <div className="received_msg">
                <div className="received_withd_msg">
                  <p>Test, which is a new approach to have</p>
                  <span className="time_date"> 11:01 AM    |    Yesterday</span>
                </div>
              </div>
            </div>
            <div className="outgoing_msg">
              <div className="sent_msg">
                <p>Apollo University, Delhi, India Test</p>
                <span className="time_date"> 11:01 AM    |    Today</span>
              </div>
            </div>
            <div className="incoming_msg">
              <div className="incoming_msg_img"><img src="https://ptetutorials.com/images/user-profile.png" alt="sunil" />
              </div>
              <div className="received_msg">
                <div className="received_withd_msg">
                  <p>We work directly with our designers and suppliers,
                    and sell direct to you, which means quality, exclusive
                    products, at a price anyone can afford.
                  </p>
                  <span className="time_date"> 11:01 AM    |    Today</span>
                </div>
              </div>
            </div>
          </div>
          <div className="type_msg">
            <div className="input_msg_write">
              <input type="text" className="write_msg" placeholder="Type a message" />
              {/* <button className="msg_send_btn" type="button"><i className="fa fa-paper-plane-o" aria-hidden="true" /></button> */}
              <Button className="msg_send_btn" type="button"><FontAwesomeIcon icon={faPaperPlane} aria-hidden="true" /></Button>
            </div>
          </div>
        </div>
      </Row>
    );

    // const { selectPlayer } = props;
    // const { props } = this;
    // const { selectPlayer, auth } = props;
    // const players = props.players && Array.isArray(props.players) ? props.players : [];

    // console.log('players', players, 'props', props);

    return (
      <Card outline color="info" className="border-info app mt-2" id="available-players" style={{ display: "block" }} >
        <CardHeader className="text-white bg-info">Chat</CardHeader>
        <CardBody>
          { blah }
        </CardBody>
        {/* <CardFooter className="card-footer">Select a user to play!</CardFooter> */}
      </Card>
    );
  }
}

function mapStateToProps({ auth, gathering }) {
  const onlinePlayers = gathering != null ? gathering.onlinePlayers || [] : [];
  return { auth, gathering, players: onlinePlayers, };
}

// export default connect(mapStateToProps, { joinGathering })(UserList);
export default connect(mapStateToProps, { changePlayerStatus })(Chat);
