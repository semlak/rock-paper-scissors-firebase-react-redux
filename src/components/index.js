import Game from './Game';
import Navbar from './Navbar';
import requireAuth from './requireAuth';
import PlayerList from './PlayerList';
import AuthenticationModal from './AuthenticationModal';
import LoginForm from './LoginForm';
import RegistrationForm from './RegistrationForm';
import Landing from './Landing';

export {
  Game,
  Navbar,
  requireAuth,
  PlayerList,
  AuthenticationModal,
  LoginForm,
  RegistrationForm,
  Landing,
};
