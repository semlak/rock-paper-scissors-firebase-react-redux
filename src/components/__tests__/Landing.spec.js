import React from 'react';
// import Enzyme, { shallow } from 'enzyme';
// import { shallow, render } from 'enzyme';
import { shallow, } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
// import configureStore from 'redux-mock-store'; // Smart components

import { Button } from 'reactstrap';

import { Landing } from '../../components/Landing';

// Enzyme.configure({ adapter: new Adapter() });

// const mockStore = configureStore();
// const initialState = {
//   auth: null,
//   modals: {
//     authenticationModalOpen: false,
//   },
//   gathering: {},
//   game: {
//     gameStatus: 'NO_GAME',
//   },
// };

// const store = mockStore(initialState);/Sna
    
// function setup() {
//   const props = {
//     toggleAuthenticationModal: jest.fn(),
//   };
//
//   const enzymeWrapper = shallow(<Landing {...props} />);
//
//   return {
//     props,
//     enzymeWrapper,
//   };
// }


describe('components', () => {
  describe('Landing', () => {
    let wrapper;
    const mockModalToggler = jest.fn();
    beforeEach(() => {
      wrapper = shallow(<Landing toggleAuthenticationModal={mockModalToggler} />);
      // wrapper = render(<Landing toggleAuthenticationModal={mockModalToggler} />);
    });

    it('should dispatch event to show Authentication modal when clicked', () => {
      // const wrapper = shallow(<Landing authenticationModalOpen={mockModalToggler}/>);

      const modalToggleButton = wrapper.find(Button);
      expect(modalToggleButton).toBeTruthy();
      modalToggleButton.simulate('click', { preventDefault: jest.fn() });
      // expect(store.getActions()).toMatchSnapshot();
      expect(mockModalToggler).toHaveBeenCalled();
    });

    it('should render to match snapshot', () => {
      const component = wrapper;
      expect(toJson(component)).toMatchSnapshot();
    });
    // it('should render self and subcomponents', () => {
    //   const { enzymeWrapper } = setup();
    //   expect(enzymeWrapper.find('div').hasClass('row')).toBe(true);
    //   expect(enzymeWrapper.find('div').hasClass('social-signin-container')).toBe(true);
    //
    //   expect(enzymeWrapper.find('a').hasClass('social-signin')).toBe(true);
    //   const modalToggleButton = enzymeWrapper.find('a.social-signin');
    //   modalToggleButton.simulate('click');
    //   // expect(modalToggleButton)
    // });
  });
});
