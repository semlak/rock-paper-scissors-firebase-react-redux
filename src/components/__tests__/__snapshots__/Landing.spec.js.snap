// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Landing should render to match snapshot 1`] = `
<div>
  <Row
    className="social-signin-container"
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <img
        alt="Rock Paper Scissors"
        id="sign-in"
        src="landing-rps.jpg"
        width="100%"
      />
      <br />
      <br />
      <br />
      <br />
      <Button
        className="social-signin"
        color="secondary"
        href="/#"
        onClick={[Function]}
        tag="button"
      >
        Sign in to start
      </Button>
    </Col>
  </Row>
  <Row
    className="mt-4 mb-4 text-left"
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Card
        body={true}
        color="danger"
        outline={true}
        tag="div"
      >
        <p>
          For trying out the game with multiple players, you can use separate logins in separate browsing sessions.
        </p>
        <p>
          You can create your own logins or use the following sample logins (each with password '
          <samp>
            password
          </samp>
          '):
        </p>
        <ul>
          <li>
            <samp>
              harry@hogwarts.co.uk
            </samp>
          </li>
          <li>
            <samp>
              ron@hogwarts.co.uk
            </samp>
          </li>
          <li>
            <samp>
              hermione@hogwarts.co.uk
            </samp>
          </li>
        </ul>
      </Card>
    </Col>
  </Row>
</div>
`;
